editor/plugins/tiles/tile_map_editor.cpp

```diff
@@ -3698,32 +3644,23 @@ void TileMapEditor::_update_layers_selection() {
 	} else {
 		tile_map_layer = -1;
 	}
-	tile_map->set_selected_layer(toogle_highlight_selected_layer_button->is_pressed() ? tile_map_layer : -1);
+	tile_map->set_selected_layer(toggle_highlight_selected_layer_button->is_pressed() ? tile_map_layer : -1);
 
-	// Build the list of layers.
-	for (int i = 0; i < tile_map->get_layers_count(); i++) {
-		String name = tile_map->get_layer_name(i);
-		layers_selection_popup->add_item(name.is_empty() ? vformat(TTR("Layer #%d"), i) : name, i);
-		layers_selection_popup->set_item_as_radio_checkable(i, true);
-		layers_selection_popup->set_item_disabled(i, !tile_map->is_layer_enabled(i));
-		layers_selection_popup->set_item_checked(i, i == tile_map_layer);
-	}
-
-	// Update the button label.
-	if (tile_map_layer >= 0) {
-		layers_selection_button->set_text(layers_selection_popup->get_item_text(tile_map_layer));
+	layers_selection_button->clear();
+	layers_empty = tile_map->get_layers_count() <= 0;
+	if (layers_empty) {
```

Comment:
```suggestion
	if (tile_map->get_layers_count() == 0) {
```
As I said above, `layers_empty` variable can be removed.